sql/core/src/main/scala/org/apache/spark/sql/execution/joins/OuterJoin.scala

```diff
@@ -63,10 +74,12 @@ trait HashOuterJoin {
     case LeftOuter => (rightKeys, leftKeys)
     case x =>
       throw new IllegalArgumentException(
-        s"HashOuterJoin should not take $x as the JoinType")
+        s"${getClass.getSimpleName} should not take $x as the JoinType")
   }
 
   protected[this] def isUnsafeMode: Boolean = {
+    // TODO(josh): there is an existing bug here: this should also check whether unsafe mode
+    // is enabled. also, the default for self.codegenEnabled looks inconsistent to me.
     (self.codegenEnabled && joinType != FullOuter
```

Comment:
This will be addressed by @davies' patch to consolidate the Unsafe and Tungsten configurations.